src/buf/fixed/pool.rs

```diff
@@ -284,6 +286,9 @@ impl<T: IoBufMut> FixedBufPool<T> {
                 return buf;
             }
 
+            // Await notify_one
+            notified.as_mut().await;
+
```

Comment:
I think the comment below clarifies why the whole branch below the `if ... { ... return buf; }` can be taken, so part of it should be placed above this.